qiskit_experiments/curve_analysis/base_curve_analysis.py

```diff
@@ -0,0 +1,544 @@
+
+    This method creates initial guesses for the fit parameters.
+    This might be overridden by subclass.
+    See :ref:`curve_analysis_init_guess` for details.
+
+    .. rubric:: _format_data
+
+    This method consumes the processed dataset and outputs the formatted dataset.
+    By default, this method takes the average of y values over
+    the same x values and then sort the entire data by x values.
+
+    .. rubric:: _evaluate_quality
+
+    This method evaluates the quality of the fit based on the fit result.
+    This returns "good" when reduced chi-squared is less than 3.0.
+    Usually it returns string "good" or "bad" according to the evaluation.
+    This criterion can be updated by subclass.
+
+    .. rubric:: _run_data_processing
+
+    This method performs data processing and returns the processed dataset.
+    By default, it internally calls :class:`DataProcessor` instance from the analysis options
+    and processes experiment data payload to create Y data with uncertainty.
+    X data and other metadata are generated within this method by inspecting the
+    circuit metadata. The series classification is also performed by based upon the
+    matching of circuit metadata and :attr:`SeriesDef.filter_kwargs`.
+
+    .. rubric:: _run_curve_fit
+
+    This method performs the fitting with predefined fit models and the formatted dataset.
+    This method internally calls :meth:`_generate_fit_guesses` method.
+    Note that this is a core functionality of the :meth:`_run_analysis` method,
+    that creates fit result object from the formatted dataset.
+
+    .. rubric:: _create_analysis_results
+
+    This method to creates analysis results for important fit parameters
+    that might be defined by analysis options ``result_parameters``.
+    In addition, another entry for all fit parameters is created when
+    the analysis option ``return_fit_parameters`` is ``True``.
+
+    .. rubric:: _create_curve_data
+
+    This method to creates analysis results for the formatted dataset, i.e. data used for the fitting.
+    Entries are created when the analysis option ``return_data_points`` is ``True``.
+    If analysis consists of multiple series, analysis result is created for
+    each curve data in the series definitions.
+
+    .. rubric:: _initialize
+
+    This method initializes analysis options against input experiment data.
+    Usually this method is called before other methods are called.
+
+    """
+
+    @property
+    @abstractmethod
+    def parameters(self) -> List[str]:
+        """Return parameters estimated by this analysis."""
+
+    @property
+    def drawer(self) -> BaseCurveDrawer:
+        """A short-cut for curve drawer instance."""
+        return self._options.curve_plotter
+
+    @classmethod
+    def _default_options(cls) -> Options:
+        """Return default analysis options.
+
+        Analysis Options:
+            curve_plotter (BaseCurveDrawer): A curve drawer instance to visualize
```

Comment:
As above comment, the option and class name should probably match, both be called drawers *or* plotters, not mixing the two. Since the class already exists and option is new I would probably change the option name to `curve_drawer` to match

Reply:
good catch! I changed option name.